### ISSUE TITLE
improve user experience

### ISSUE THREAD

**Issue by pointhi** [searx/searx#226]
searx is greate, but there are some things where new users get problems. I think we should collect those issues and thinking for sollutions.

This are my known problems for new users:
- they don't know about the preferences (probably using icons)
- they ask why there is no autocompletion
- they ask why all results are in english, and not in there own language
- selecting categories is different as in all the other search engines

If possible, we should use a friend or someone from our family to try searx it's first time without help, and note problems which are occour. Like in the following commic :):

![Houswife Test](http://www.commitstrip.com/wp-content/uploads/2014/01/Strips-Le-test-de-ta-m%C3%A8re-650-finalenglish.jpg)

based on the comment from @Cqoicebordel in #222

**Comment by dimqua** [comment]
My two cents:
- Duplicate results should be combined and placed on top. (like Ixquick do)
- Categories like "Images" enable too many (all) engines by default. As result, searx isn't meeting expectations of new users by providing result from uncommon/unusual sources.

**Comment by dalf** [comment]
> they ask why all results are in English, and not in there own language

Is it the issue #77 ( value based on the "Accept Languages" HTTP header) ?

In my opinion the value "Automatic" in "Search language" preference is misleading since it's really mean no language preference or worse English language. In fact in most of the engine, there is something like

``` python
    if params['language'] == 'all':
        language = 'en'
```

BTW the internal value is "all" which is different from user value "Automatic".

Moreover, even it is not the case for now, what to do if an engine has "language_support = False" ? If the users doesn't set "Search language" to "all", the engine can't use ( https://github.com/asciimoo/searx/blob/master/searx/search.py#L450 and https://github.com/asciimoo/searx/blob/master/searx/engines/__init__.py#L71 ). What is the use case of this feature ?

**Comment by asciimoo** [comment]
@dalf, you're right, language handling is a bit messy. The reason to set `language_support` to `True` by default is only historical - there wasn't enough engine with language support.

**Comment by Cqoicebordel** [comment]
I asked for this kind of help on FB, and for now, they think that it's a bit weird to have to click again on the magnifier glass after clicking on the images category (and also to have to deactivate General).  
I think that what is missing is the checkbox euphemism. Everybody know how a checkbox works, but the categories don't have that little square box with a tick in it. Maybe just by acknowledging that they are checkboxes, it would simplify the user's understanding of their mechanics.

**Comment by dalf** [comment]
What you say, it's that the common use case is a search with one category, at least for images.

That's why, I suggest that searx records the categories choices and frequency for searches (frequency in percentage to preserve privacy).

May be, it could help (but not decide) to choose the right category / sub category ; and UI / UX.

About the design : why not ask help on twitter ? or feedback (but not before there is a forum)

**Comment by Cqoicebordel** [comment]
No, I'm saying that they expected the search on images to be launched at the precise moment they clicked on Images.

**Comment by pointhi** [comment]
Probably, I can add an option in the oscar-template to allow search only in one category.

**Comment by Cqoicebordel** [comment]
I don't think it will solve the issue : it's for the first time users, and obviously, for them, an option won't change anything.  
I think it's only solvable in an UI way. We must show that the categories buttons are switches and not tabs.  
It could be done by changing the color from blue and white to green and red, or using an image of a switch, or... I don't know :)

**Comment by pointhi** [comment]
probably we could add a Check mark if the engine is selected.

**Comment by JocelynDelalande** [comment]
> selecting categories is different as in all the other search engines

Yes, to expand more on that,  (@pointhi, stop me if wrong) the fact you can select several categories at once by default is confusing, it's very-power-user to search images _and_ files _and_ web at once, so it's a lot of clicks to change which category you are searching in.

This one is annoying with all casual users I had to test my searx instance. As a matter of fact, the non-profit [framasoft](http://framasoft.net) is hosting a big instance with customized-ui version of searx, they changed this behaviour: https://framabee.org/

**Comment by dalf** [comment]
A brainstorrming / cut & paste from other search engines.

![brainstorming ui](https://cloud.githubusercontent.com/assets/1594191/21369964/cd496746-c709-11e6-931d-be3d6eb7766f.png)

* Seems to uses the space more efficiently, no need to click on "Advanced settings" 
* Preferences / About could be with a menu on the right (can be implemented with only CSS)
* Number of result could be on the left column just before the first result
* The choice of color is ... 
* The button on infoboxes and suggestion becomes link  

No HTML sorry : it's a messy edit of the templates, HTML using the Inspector and Gimp.

**Comment by kvch** [comment]
@dalf Looks cool. However, it is very similar to DDG. But I am not sure if it is a problem. :)

**Comment by kvch** [comment]
I've just started to work in realizing @dalf's mock.
